lib/matplotlib/axes/_axes.py

```diff
@@ -3317,6 +3317,10 @@ def get_next_color():
                 else:
                     raise TypeError(
                         'autopct must be callable or a format string')
+                
+                if mpl.rcParams["text.usetex"] or 'usetex' in textprops:
+                    s = s.replace("%", "\%")
```

Comment:
```suggestion
                    s = s.replace("%", r"\%")
```

The `\%` is invalid and causing flake8 (and python 3.12) issues